pkg/flist/cleanup.go

```diff
@@ -139,9 +139,32 @@ func (f *flistModule) cleanUnusedMounts() error {
 		}
 
 		if err := os.Remove(path); err != nil {
-			log.Error().Err(err).Msgf("failed to clean mountpoint %s", path)
+			log.Debug().Msgf("failed to remove path, trying to forcibly clean up : %+v", path)
+			if err := f.forceUnmountAndRemove(path); err != nil {
+				log.Error().Err(err).Msgf("failed to clean mountpoint %s", path)
+			}
 		}
 	}
 
 	return nil
 }
+
+// forceUnmountAndRemove tries to fix the clean up of broken mounts by attempting to unmount them first
+func (f *flistModule) forceUnmountAndRemove(path string) error {
+	// try normal unmount first
+	err := f.system.Unmount(path, 0)
+	if err != nil {
+		log.Warn().Err(err).Msgf("normal unmount failed for %s, trying lazy unmount", path)
+
+		// try lazy unmount
+		err = f.system.Unmount(path, syscall.MNT_DETACH)
```

Comment:
this is not force unmount that is lazy unmount which is something diff from the method name 
so I suggest u try to unmount if not try force unmount if u really want to force unmount 
`err := syscall.Unmount("/mnt/my-mount", syscall.MNT_FORCE)`
or change method name to reflect u r doing lazy unmount but sure this needs some testing to check if force unmount will slove it or not